app/v1/models/OriginData.scala

```diff
@@ -0,0 +1,109 @@
+      false
+    }
+  }
+
+  private[models] def countryCodeValidation: Option[Int] => Boolean = {
+    case Some(countryCode) =>
+      val passedValidation: Boolean = (countryCode >= 0) && (countryCode <= 286)
+      if (!passedValidation) Logger.warn("[OriginData][countryCodeValidation] - country code is not valid")
+      passedValidation
+    case _ => true
+  }
+
+
+  implicit val reads: Reads[OriginData] = (
+    birthTownPath.readNullable[String]
+      .filter(JsonValidationError("Birth town does not match regex"))(birthTown => stringValidation(birthTown, "birth town")) and
+      birthProvincePath.readNullable[String]
+        .filter(JsonValidationError("Birth province does not match regex"))(birthProvince => stringValidation(birthProvince, "birth town")) and
+      birthCountryCodePath.readNullable[Int]
+        .filter(JsonValidationError("Country code is not valid"))(countryCodeValidation) and
+      nationalityPath.readNullable[Int]
+        .filter(JsonValidationError("Nationality code is not valid"))(countryCodeValidation) and
+      birthSurnamePath.readNullable[String]
+        .filter(JsonValidationError("Birth surname does not match regex"))(surname => stringValidation(surname, "birth surname")) and
+      maternalForenamePath.readNullable[String]
+        .filter(JsonValidationError("Maternal forename does not match regex"))(forename => stringValidation(forename, "maternal forename")) and
+      maternalSurnamePath.readNullable[String]
+        .filter(JsonValidationError("Maternal surname does not match regex"))(surname => stringValidation(surname, "maternal surname")) and
+      paternalForenamePath.readNullable[String]
+        .filter(JsonValidationError("Paternal forename does not match regex"))(forename => stringValidation(forename, "paternal forename")) and
+      paternalSurnamePath.readNullable[String]
+        .filter(JsonValidationError("Paternal surname does not match regex"))(surname => stringValidation(surname, "paternal surname")) and
+      foreignSocialSecurityPath.readNullable[String]
+        .filter(JsonValidationError("Foreign social security does not match regex"))(socialSecurity => stringValidation(socialSecurity, "birth surname")) and
+      lastEUAddressPath.readNullable[LastEUAddress]
+    ) (OriginData.apply _)
+
+  implicit val writes: Writes[OriginData] = (
+    birthTownPath.writeNullable[String] and
+      birthProvincePath.writeNullable[String] and
+      birthCountryCodePath.writeNullable[Int] and
+      nationalityPath.writeNullable[Int] and
+      birthSurnamePath.writeNullable[String] and
+      maternalForenamePath.writeNullable[String] and
+      maternalSurnamePath.writeNullable[String] and
+      paternalForenamePath.writeNullable[String] and
+      paternalSurnamePath.writeNullable[String] and
+      foreignSocialSecurityPath.writeNullable[String] and
+      lastEUAddressPath.writeNullable[LastEUAddress]
+    ) (unlift(OriginData.unapply))
```

Comment:
Isn't writing this out manually redundant? `Json.writes[OriginData]` would achieve the same thing, no?